### PR TITLE
Add codemodder entry point script

### PR DESCRIPTION
**Pull request by drdavella**
## Overview
*Add `codemodder` entry point script*

## Description

* This is mainly motivated by the need to integrate with the codemodder orchestrator
* Eventually I want to deprecate the `python -m codemodder` CLI, which I think will allow us to simplify a lot of the code around argument parsing, etc.

### CODE REVIEW

File: src/codemodder/__main__.py [hunk 1]
```diff
@@ -192,6 +192,14 @@ def run(argv, original_args) -> int:
     return 0
 
 
-if __name__ == "__main__":
+def run_script():
+    # TODO: I'm not sure why this needs to be parsed out separately
+    # Maybe it has something to do with the invocation as python -m codemodder.
```
Comment (clavedeluna) [inline src/codemodder/__main__.py:197]: IIRC it's to make testing easier, which isn't the best reason in the world. I do think we should go ahead and deprecate this and update the tests, but if you want to do that later that's cool with me.

File: src/codemodder/__main__.py [hunk 2]
```diff
@@ -192,6 +192,14 @@ def run(argv, original_args) -> int:
     return 0
 
 
-if __name__ == "__main__":
+def run_script():
```
Comment (clavedeluna) [inline src/codemodder/__main__.py]: what about calling it `main`? don't love the wording with script

**Review by clavedeluna — APPROVED** [review]

File: src/codemodder/__main__.py (hunk 2 above)
Comment (drdavella) [reply src/codemodder/__main__.py]: Yeah that's a better idea.

File: src/codemodder/__main__.py (hunk 1 above)
Comment (drdavella) [reply src/codemodder/__main__.py:197]: I will update the tests 👍

**Pull request merged**